clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy

```diff
@@ -808,11 +808,19 @@ class KubernetesApiConverter {
         builder = builder.withNewRollingUpdate()
 
         if (rollingUpdate.maxSurge) {
-          builder = builder.withNewMaxSurge(rollingUpdate.maxSurge)
+          def maxSurge = rollingUpdate.maxSurge
```

Comment:
I didn't chase down the api docs, but I'm guessing int is preferred, but either int or string is ok?
Maybe add a comment explaining why the extra complication here, including what exception types are possible (guessing NPE and NFE).

Or you could pull this logic out into a method that returns an `IntOrString` and use it here and below.